ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -134,36 +141,50 @@ public void shutdown() {
   }
 
   @Override
-  public void handleGet(RestRequest restRequest, RestResponseChannel restResponseChannel) {
+  public void handleGet(final RestRequest restRequest, final RestResponseChannel restResponseChannel) {
     long processingStartTime = System.currentTimeMillis();
     long preProcessingTime = 0;
     handlePrechecks(restRequest, restResponseChannel);
     try {
       logger.trace("Handling GET request - {}", restRequest.getUri());
       checkAvailable();
       RestUtils.SubResource subresource = RestUtils.getBlobSubResource(restRequest);
-      RestRequestMetrics requestMetrics =
-          restRequest.getSSLSession() != null ? frontendMetrics.getBlobSSLMetrics : frontendMetrics.getBlobMetrics;
-      if (subresource != null) {
-        logger.trace("Sub-resource requested: {}", subresource);
-        switch (subresource) {
-          case BlobInfo:
-            requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getBlobInfoSSLMetrics
-                : frontendMetrics.getBlobInfoMetrics;
-            break;
-          case UserMetadata:
-            requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getUserMetadataSSLMetrics
-                : frontendMetrics.getUserMetadataMetrics;
-            break;
+      String operationOrBlobId =
+          RestUtils.getOperationOrBlobIdFromUri(restRequest, subresource, frontendConfig.frontendPathPrefixesToRemove);
+      if (operationOrBlobId.startsWith("/")) {
+        operationOrBlobId = operationOrBlobId.substring(1);
```

Comment:
It is sometimes necessary to keep the leading slash in case of blob ids. So I think it is better to not to remove the leading slash inside `getOperationOrBlobIdFromUr()i` but let the caller strip it if they deem it safe (or send "/" as one of the prefixes to remove).